controllers/control/jobs.go

```diff
@@ -84,9 +84,16 @@ func (r *CassandraTaskReconciler) restartSts(ctx context.Context, sts []appsv1.S
 				status.ReadyReplicas == status.Replicas &&
 				status.ObservedGeneration == st.GetObjectMeta().GetGeneration() {
 				// This one has been updated, move on to the next one
+
+				restartedPods[st.Name] = int(status.UpdatedReplicas)
 				continue
 			}
-
+			restartedPods[st.Name] = int(status.UpdatedReplicas)
+			totalRestarted := 0
+			for _, v := range restartedPods {
```

Comment:
This code is never called with finished StSes as they jump out of the loop before this is reached.

Reply:
I know, but we want unfinished ones to be tracked too, don't we?

Reply:
We do, but now we won't track the finished ones at all. We should update the TaskConfig.Completed inside that part of the loop also. Or use a pointer and update that one in TaskConfig..

Reply:
Can you check my latest commit and confirm that's what you're after?